plugins/movetoarchive.koplugin/main.lua

```diff
@@ -106,12 +106,8 @@ function MoveToArchive:commonProcess(is_move_process, moved_done_text)
         FileManager:copyRecursive(DocSettings:getSidecarDir(document_full_path), self.archive_dir_path)
     end
     local dest_file = string.format("%s%s", self.archive_dir_path, filename)
-    ReadHistory:updateItemByPath(document_full_path, dest_file)
+    ReadHistory:updateItemByPath(document_full_path, dest_file) -- (will update "lastfile" if needed)
     ReadCollection:updateItemByPath(document_full_path, dest_file)
```

Comment:
`ReadHistory:updateItemByPath()` updates history and what does 
`ReadCollection:updateItemByPath()` ? 
I glanced at file but didn't understand what's it's doing. 

Isn't it something that should be also done in `ReadHistory`? If we also have to do this toughter?

Reply:
ReadCollection is another bit of code that manages what's called "Favorites" in the UI.
ReadHistory doesn't know about it - and should not.
So, unless we add a wrapper module `HouseKeeper` and a `HouseKeeper:updateItemByPath(old, new)` that would call this on all the relevant modules, we'll do with it :)
(Or events, but not all modules might be registered to UIManager and get that event.)